tls/extensions/s2n_client_key_share.c

```diff
@@ -153,13 +156,16 @@ int s2n_extensions_client_key_share_send(struct s2n_connection *conn, struct s2n
 static int s2n_ecdhe_supported_curves_send(struct s2n_connection *conn, struct s2n_stuffer *out)
 {
     notnull_check(conn);
+    notnull_check(conn->config);
+    const struct s2n_ecc_preferences *ecc_pref = conn->config->ecc_preferences;
+    notnull_check(ecc_pref);
 
     const struct s2n_ecc_named_curve *named_curve = NULL;
     struct s2n_ecc_evp_params *ecc_evp_params = NULL;
 
-    for (uint32_t i = 0; i < s2n_ecc_evp_supported_curves_list_len; i++) {
+    for (uint32_t i = 0; i < ecc_pref->count; i++) {
```

Comment:
lets have an issue to track sending selective keyshares.

Reply:
@rday  is working on these changes in his PR.

Reply:
This will be addressed in the HelloRetry feature. We need support for generating a single key share when responding to a retry request, and it isn't hard to update this code to only add the highest priority key share in the preference list.